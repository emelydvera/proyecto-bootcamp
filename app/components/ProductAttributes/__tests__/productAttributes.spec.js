const React = require("react");
const ProductAttributes = require("../ProductAttributes");
const { render, screen } = require("@testing-library/react");
const { queryByText } = screen;
const mockAttributes = require("../../../../mocks/components/productAttributes/attributes.json");

describe("ProductAttribues component", () => {
  let component;
  const i18n = { gettext: (text) => text };

  beforeEach(() => {
    component = render(
      <ProductAttributes i18n={i18n} attributes={mockAttributes} />
    );
  });

  it("should render component", () => {
    const { asFragment } = component;
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render text title 'Características'", () => {
    const tableTitle = queryByText("Características");
    expect(tableTitle).toBeInTheDocument();
  });

  it("should render product model attribute and its value", () => {
    const model = queryByText("Modelo alfanumérico");
    const modelValue = queryByText("15ALC6");

    expect(model).toBeInTheDocument();
    expect(modelValue).toBeInTheDocument();
  });

  it("should render product brand attribute and its value", () => {
    const brand = queryByText("Marca");
    const brandValue = queryByText("Lenovo");

    expect(brand).toBeInTheDocument();
    expect(brandValue).toBeInTheDocument();
  });

  it("should render product detailed model attribute and its value", () => {
    const detailedModel = queryByText("Modelo detallado");
    const detailedModelValue = queryByText("82KU00AAUS");

    expect(detailedModel).toBeInTheDocument();
    expect(detailedModelValue).toBeInTheDocument();
  });
});
