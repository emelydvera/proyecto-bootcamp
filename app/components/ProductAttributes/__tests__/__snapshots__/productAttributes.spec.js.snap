// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductAttribues component should render component 1`] = `
<DocumentFragment>
  <details
    class="attributes"
  >
    <summary
      class="attributes__title"
      tabindex="23"
    >
      Características
    </summary>
    <table
      class="attributes__table"
    >
      <tbody>
        <tr
          tabindex="23"
        >
          <td>
            Modelo alfanumérico
          </td>
          <td>
            15ALC6
          </td>
        </tr>
        <tr
          tabindex="23"
        >
          <td>
            Marca
          </td>
          <td>
            Lenovo
          </td>
        </tr>
        <tr
          tabindex="23"
        >
          <td>
            Modelo detallado
          </td>
          <td>
            82KU00AAUS
          </td>
        </tr>
      </tbody>
    </table>
  </details>
</DocumentFragment>
`;
